File: ClassReminderWeb/src/models/AppTheme.js
/**
 * 앱 테마/스킨 설정
 */
export const AppThemes = {
  DARK_HORROR: {
    name: 'DARK_HORROR',
    displayName: '다크 호러',
    emoji: '🦇',
    primaryColor: '#8B0000',
    secondaryColor: '#2C0000',
    backgroundColor: '#1A0000',
    textColor: '#FFFFFF',
    cardBackground: '#2C0000'
  },

  NEON_CYBER: {
    name: 'NEON_CYBER',
    displayName: '네온 사이버',
    emoji: '⚡',
    primaryColor: '#FF00FF',
    secondaryColor: '#00FFFF',
    backgroundColor: '#0A0A1F',
    textColor: '#FFFFFF',
    cardBackground: '#1A1A3F'
  },

  PASTEL_DREAM: {
    name: 'PASTEL_DREAM',
    displayName: '파스텔 드림',
    emoji: '🌸',
    primaryColor: '#FFB3BA',
    secondaryColor: '#BAFFC9',
    backgroundColor: '#FFFFBA',
    textColor: '#333333',
    cardBackground: '#FFFFFF'
  },

  OCEAN_BLUE: {
    name: 'OCEAN_BLUE',
    displayName: '오션 블루',
    emoji: '🌊',
    primaryColor: '#1E88E5',
    secondaryColor: '#0D47A1',
    backgroundColor: '#E3F2FD',
    textColor: '#000000',
    cardBackground: '#FFFFFF'
  },

  FOREST_GREEN: {
    name: 'FOREST_GREEN',
    displayName: '포레스트 그린',
    emoji: '🌲',
    primaryColor: '#388E3C',
    secondaryColor: '#1B5E20',
    backgroundColor: '#E8F5E9',
    textColor: '#000000',
    cardBackground: '#FFFFFF'
  },

  SUNSET_ORANGE: {
    name: 'SUNSET_ORANGE',
    displayName: '선셋 오렌지',
    emoji: '🌅',
    primaryColor: '#FF6F00',
    secondaryColor: '#E65100',
    backgroundColor: '#FFF3E0',
    textColor: '#000000',
    cardBackground: '#FFFFFF'
  },

  GALAXY_PURPLE: {
    name: 'GALAXY_PURPLE',
    displayName: '갤럭시 퍼플',
    emoji: '🌌',
    primaryColor: '#7B1FA2',
    secondaryColor: '#4A148C',
    backgroundColor: '#F3E5F5',
    textColor: '#000000',
    cardBackground: '#FFFFFF'
  },

  RETRO_GAME: {
    name: 'RETRO_GAME',
    displayName: '레트로 게임',
    emoji: '🎮',
    primaryColor: '#00FF00',
    secondaryColor: '#FFFF00',
    backgroundColor: '#000000',
    textColor: '#00FF00',
    cardBackground: '#1A1A1A'
  },

  CANDY_POP: {
    name: 'CANDY_POP',
    displayName: '캔디 팝',
    emoji: '🍭',
    primaryColor: '#FF69B4',
    secondaryColor: '#FF1493',
    backgroundColor: '#FFF0F5',
    textColor: '#000000',
    cardBackground: '#FFFFFF'
  },

  MINIMALIST: {
    name: 'MINIMALIST',
    displayName: '미니멀리스트',
    emoji: '⚪',
    primaryColor: '#424242',
    secondaryColor: '#757575',
    backgroundColor: '#FAFAFA',
    textColor: '#212121',
    cardBackground: '#FFFFFF'
  }
};

/**
 * 모든 테마 목록 가져오기
 */
export function getAllThemes() {
  return Object.values(AppThemes);
}

/**
 * 테마 이름으로 테마 가져오기
 */
export function getTheme(themeName) {
  return AppThemes[themeName] || AppThemes.OCEAN_BLUE;
}

File: ClassReminderApp/app/src/main/java/com/classreminder/models/AppTheme.kt
package com.classreminder.models

/**
 * 앱 테마/스킨 설정
 */
enum class AppTheme(
    val displayName: String,
    val primaryColor: String,
    val secondaryColor: String,
    val backgroundColor: String,
    val textColor: String,
    val emoji: String
) {
    DARK_HORROR(
        "다크 호러",
        "#8B0000",
        "#2C0000",
        "#1A0000",
        "#FFFFFF",
        "🦇"
    ),

    NEON_CYBER(
        "네온 사이버",
        "#FF00FF",
        "#00FFFF",
        "#0A0A1F",
        "#FFFFFF",
        "⚡"
    ),

    PASTEL_DREAM(
        "파스텔 드림",
        "#FFB3BA",
        "#BAFFC9",
        "#FFFFBA",
        "#333333",
        "🌸"
    ),

    OCEAN_BLUE(
        "오션 블루",
        "#1E88E5",
        "#0D47A1",
        "#E3F2FD",
        "#000000",
        "🌊"
    ),

    FOREST_GREEN(
        "포레스트 그린",
        "#388E3C",
        "#1B5E20",
        "#E8F5E9",
        "#000000",
        "🌲"
    ),

    SUNSET_ORANGE(
        "선셋 오렌지",
        "#FF6F00",
        "#E65100",
        "#FFF3E0",
        "#000000",
        "🌅"
    ),

    GALAXY_PURPLE(
        "갤럭시 퍼플",
        "#7B1FA2",
        "#4A148C",
        "#F3E5F5",
        "#000000",
        "🌌"
    ),

    RETRO_GAME(
        "레트로 게임",
        "#00FF00",
        "#FFFF00",
        "#000000",
        "#00FF00",
        "🎮"
    ),

    CANDY_POP(
        "캔디 팝",
        "#FF69B4",
        "#FF1493",
        "#FFF0F5",
        "#000000",
        "🍭"
    ),

    MINIMALIST(
        "미니멀리스트",
        "#424242",
        "#757575",
        "#FAFAFA",
        "#212121",
        "⚪"
    );

    companion object {
        fun fromString(value: String): AppTheme {
            return values().find { it.name == value } ?: OCEAN_BLUE
        }
    }

    /**
     * 테마의 주요 색상을 Int로 반환
     */
    fun getPrimaryColorInt(): Int {
        return android.graphics.Color.parseColor(primaryColor)
    }

    fun getSecondaryColorInt(): Int {
        return android.graphics.Color.parseColor(secondaryColor)
    }

    fun getBackgroundColorInt(): Int {
        return android.graphics.Color.parseColor(backgroundColor)
    }

    fun getTextColorInt(): Int {
        return android.graphics.Color.parseColor(textColor)
    }
}
